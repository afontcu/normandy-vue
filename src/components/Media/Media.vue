<template>
  <div class="o-media"
    :class="[
      align && `o-media--${align}`,
      size && `o-media--${size}`,
      { 'o-media--reverse': reverse }
    ]">

    <div class="o-media__img">
      <img v-bind="{ src, alt }" />
    </div>

    <div class="o-media__body">
      <slot></slot>
    </div>

  </div>
</template>

<script>
const VALIGN_VALUES = [ 'center', 'bottom', 'stretch' ]
const SIZE_VALUES = [ 'flush', 'tiny', 'small', 'large', 'huge' ]

export default {
  name: 'Media',
  props: {
    src: {
      type: String,
      required: true
    },
    alt: {
      type: String,
      required: true
    },
    reverse: {
      type: Boolean,
      default: false
    },
    size: {
      type: String,
      validator: value => SIZE_VALUES.includes(value)
    },
    align: {
      type: String,
      validator: value => VALIGN_VALUES.includes(value)
    }
  },
  data () {
    return {}
  }
}
</script>
