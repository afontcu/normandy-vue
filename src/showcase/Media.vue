<template>
  <div>
    <h1>Media Object</h1>
    <hr />
    <p>{{ attributes.description }}</p>

    <div v-for="(prop, propName) in attributes.props">
      <h2>{{ propName }} modifiers</h2>
      <hr>
      <p v-if="prop.description">{{ prop.description }}</p>
      <pre>prop: <b>{{ propName }}</b> ({{ prop.type }})</pre>

      <div class="object-wrapper" v-for="(computedProp, index) in computedProps(propName)">
        <media
          :key="index"
          v-bind="computedProp">
          {{ attributes.slot }}
        </media>
      </div>
    </div>
<!--

    <h2>Alignment modifiers</h2>
    <pre><b>prop</b>: align (String)
<b>values</b>: ['center', 'bottom']</pre>
    <hr />
    <p>Center</p>
    <media src="http://via.placeholder.com/350x150" alt="logo" align="center">
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Numquam, reiciendis fugiat ut voluptate fugit quisquam modi maiores, perspiciatis quasi praesentium ab vel at repellat amet atque? Veniam, officia, sint.
    </media>

    <p>Bottom</p>
    <media src="http://via.placeholder.com/350x150" alt="logo" align="bottom">
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Numquam, reiciendis fugiat ut voluptate fugit quisquam modi maiores, perspiciatis quasi praesentium ab vel at repellat amet atque? Veniam, officia, sint.
    </media>

    <p>Stretch</p>
    <media src="http://via.placeholder.com/350x150" alt="logo" align="stretch">
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Numquam, reiciendis fugiat ut voluptate fugit quisquam modi maiores, perspiciatis quasi praesentium ab vel at repellat amet atque? Veniam, officia, sint.
    </media>


    <h2>Reverse modifier</h2>
    <pre><b>prop</b>: reverse (Boolean)</pre>
    <hr />
    <p>Reverse the direction</p>

    <media src="http://via.placeholder.com/350x150" alt="logo" reverse>
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Numquam, reiciendis fugiat ut voluptate fugit quisquam modi maiores, perspiciatis quasi praesentium ab vel at repellat amet atque? Veniam, officia, sint.
    </media>-->
  </div>
</template>

<script>
import MediaObject from '@/components/Media'
import { MediaAttrs } from './componentsList'

export default {
  components: {
    [MediaObject.name]: MediaObject
  },
  data () {
    return {
      attributes: MediaAttrs
    }
  },
  methods: {
    computedProps (currentProp) {
      let props = []
      this.attributes.props[currentProp].values.map(value => {
        props.push({
          [currentProp]: value,
          ...this.attributes.requiredProps
        })
      })

      return props
    }
  }
}
</script>

<style></style>
