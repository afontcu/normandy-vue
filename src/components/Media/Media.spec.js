import Vue from 'vue'
import Media from './Media'

const Constructor = Vue.extend(Media)
const vm = new Constructor({
  propsData: {
    src: 'http://example.com',
    alt: 'alt'
  }
})

describe('Media.vue', () => {
  it('should be able to mount itself', () => vm.$mount())
  it('should match the snapshot', () => expect(vm.$mount().$el).toMatchSnapshot())
})
