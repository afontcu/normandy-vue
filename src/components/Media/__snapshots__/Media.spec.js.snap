// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Media.vue should match the snapshot 1`] = `
<div
  class="o-media"
>
  <div
    class="o-media__img"
  >
    <img
      src="http://example.com"
      alt="alt"
    />
  </div>
  <div
    class="o-media__body"
  />
</div>
`;
